src/main/java/dev/jbang/source/GroovyScriptSource.java

```diff
@@ -0,0 +1,60 @@
+package dev.jbang.source;
+
+import static dev.jbang.net.GroovyManager.resolveInGroovyHome;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.jboss.jandex.ClassInfo;
+
+import dev.jbang.cli.BaseBuildCommand;
+import dev.jbang.net.GroovyManager;
+
+public class GroovyScriptSource extends ScriptSource {
+
+	protected GroovyScriptSource(ResourceRef script, Function<String, String> replaceProperties) {
+		super(script, replaceProperties);
+	}
+
+	@Override
+	public List<String> getCompileOptions() {
+		return Collections.emptyList();
+	}
+
+	@Override
+	public String getCompilerBinary(String requestedJavaVersion) {
+		return resolveInGroovyHome("groovyc", getGroovyVersion());
+	}
+
+	@Override
+	public Predicate<ClassInfo> getMainFinder() {
+		return pubClass -> pubClass.method("main", BaseBuildCommand.STRINGARRAYTYPE) != null
+				|| pubClass.method("main") != null;
+	}
+
+	@Override
+	public List<String> getAllDependencies() {
+		final List<String> allDependencies = super.getAllDependencies();
+		final String groovyVersion = getGroovyVersion();
+		if (groovyVersion.startsWith("4.")) {
+			allDependencies.add("org.apache.groovy:groovy:" + groovyVersion);
```

Comment:
why is this one added ? is it as a "should always be there" kind of dependency?

Reply:
Yes, if you want to run Java class compiled by groovyc, and you should add groovy-version.jar to include Groovy APIs
and Groovy Development Kit enhancements. 

```
❯ javap Hello
Compiled from "Hello.groovy"
public class Hello extends groovy.lang.Script {
  public static transient boolean __$stMC;
  public Hello();
  public Hello(groovy.lang.Binding);
  public static void main(java.lang.String...);
  public java.lang.Object run();
  protected groovy.lang.MetaClass $getStaticMetaClass();
}
```